Libraries/Opc.Ua.Server/Subscription/MonitoredItem.cs

```diff
@@ -1079,6 +1079,89 @@ public virtual void QueueEvent(EventFieldList fields)
             }
         }
 
+        /// <summary>
+        /// Determines whether an event can be sent with SupportsFilteredRetain in consideration.
+        /// </summary>
+        /// <returns></returns>
+        protected bool CanSendFilteredAlarm(FilterContext context, EventFilter filter, IFilterTarget instance)
+        {
+            bool passedFilter = filter.WhereClause.Evaluate(context, instance);
+
+            ConditionState alarmCondition = null;
+            NodeId conditionId = null;
+            InstanceStateSnapshot instanceStateSnapshot = instance as InstanceStateSnapshot;
+            if (instanceStateSnapshot != null)
+            {
+                alarmCondition = instanceStateSnapshot.Handle as ConditionState;
+
+                if (alarmCondition != null &&
+                    alarmCondition.SupportsFilteredRetain != null &&
+                    alarmCondition.SupportsFilteredRetain.Value &&
+                    filter.SelectClauses != null)
+                {
+                    int lastIndex = filter.SelectClauses.Count - 1;
```

Comment:
Does lastIndex allways point to the ConditionId ? Can't the client request it in another order together with other SelectClauses?